network-upgrader/src/main.rs

```diff
@@ -217,17 +239,41 @@ async fn upload(upload: &Upload) -> anyhow::Result<()> {
         wallet.add_witnesses(&mut builder)?;
         wallet.adjust_for_fee(&mut builder, 0).await?;
         let tx = builder.build(provider).await?;
-
-        let result = provider.send_transaction(tx).await?;
-        println!("Subsection `{i}` successfully committed to the network with tx id `{result}`.");
-        // Wait for sentries to update off-chain database.
-        tokio::time::sleep(Duration::from_secs(4)).await;
+        let tx_id = tx.id(chain_id);
+
+        let result = provider.send_transaction_and_await_commit(tx).await?;
+
+        match result {
+            TxStatus::Success { .. } => {
+                println!(
+                    "Subsection `{i}` successfully committed to the network with tx id `{tx_id}`."
+                );
+            }
+            TxStatus::Submitted => {
+                bar.abandon_with_message("an error was detected");
+                anyhow::bail!(
+                    "The transaction `{tx_id}` is submitted and awaiting confirmation."
+                );
+            }
+            TxStatus::SqueezedOut { .. } => {
+                bar.abandon_with_message("an error was detected");
+                anyhow::bail!("Subsection `{i}` upload failed. The transaction `{tx_id}` was squeezed out.");
+            }
+            TxStatus::Revert { reason, .. } => {
+                bar.abandon_with_message("an error was detected");
+                anyhow::bail!("Subsection `{i}` upload failed. The transaction `{tx_id}` was reverted with reason `{reason}`.");
+            }
+        }
```

Comment:
Nit: Generally error messages should be lowercased.

> Error messages are typically concise lowercase sentences without trailing punctuation:
See https://doc.rust-lang.org/std/error/trait.Error.html

```suggestion
            TxStatus::Submitted => {
                bar.abandon_with_message("an error was detected");
                anyhow::bail!(
                    "the transaction `{tx_id}` is submitted and awaiting confirmation."
                );
            }
            TxStatus::SqueezedOut { .. } => {
                bar.abandon_with_message("an error was detected");
                anyhow::bail!("subsection `{i}` upload failed. The transaction `{tx_id}` was squeezed out.");
            }
            TxStatus::Revert { reason, .. } => {
                bar.abandon_with_message("an error was detected");
                anyhow::bail!("subsection `{i}` upload failed. The transaction `{tx_id}` was reverted with reason `{reason}`.");
            }
        }
```